packages/full-stack-tests/src/hierarchy-builder/grouping/LabelGrouping.test.ts

```diff
@@ -0,0 +1,360 @@
+                    instanceKeys: [keys.childSubject1],
+                    children: false,
+                  }),
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject3],
+                    children: false,
+                  }),
+                ],
+              }),
+              NodeValidators.createForLabelGroupingNode({
+                label: labelGroupName2,
+                children: [
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject2],
+                    children: false,
+                  }),
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject4],
+                    children: false,
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+      });
+    });
+
+    describe("Label and Class grouping", () => {
+      it("first groups by class then by label", async function () {
+        const labelGroupName1 = "test1";
+        const labelGroupName2 = "test2";
+        const { imodel, ...keys } = await buildIModel(this, async (builder) => {
+          const rootSubject = { className: subjectClassName, id: IModel.rootSubjectId };
+          const childSubject1 = insertSubject({ builder, codeValue: "A1", parentId: rootSubject.id, userLabel: labelGroupName1 });
+          const childSubject2 = insertSubject({ builder, codeValue: "A2", parentId: rootSubject.id, userLabel: labelGroupName1 });
+          const childPartition3 = insertPhysicalPartition({ builder, codeValue: "B3", parentId: rootSubject.id, userLabel: labelGroupName1 });
+          const childPartition4 = insertPhysicalPartition({ builder, codeValue: "B4", parentId: rootSubject.id, userLabel: labelGroupName2 });
+          const childPartition5 = insertPhysicalPartition({ builder, codeValue: "B5", parentId: rootSubject.id, userLabel: labelGroupName2 });
+          return { rootSubject, childSubject1, childSubject2, childPartition3, childPartition4, childPartition5 };
+        });
+
+        const customHierarchy: IHierarchyLevelDefinitionsFactory = {
+          async defineHierarchyLevel(parentNode) {
+            if (!parentNode) {
+              return [
+                {
+                  fullClassName: subjectClassName,
+                  query: {
+                    ecsql: `
+                              SELECT ${await selectClauseFactory.createSelectClause({
+                                ecClassId: { selector: `this.ECClassId` },
+                                ecInstanceId: { selector: `this.ECInstanceId` },
+                                nodeLabel: "root subject",
+                              })}
+                              FROM ${subjectClassName} AS this
+                              WHERE this.ECInstanceId = (${IModel.rootSubjectId})
+                            `,
```

Comment:
please check & fix template literal strings' indentation in this test

Reply:
Fixed indentation

Reply:
As far as I can tell it's still not okay in LabelGrouping and MultiLevelGrouping tests

Reply:
I think I did it now.